### DISCUSSION TITLE
running on HPC3

### DISCUSSION THREAD

**Discussion by Aquios7** [BLAST-WarpX/warpx#4697] [Q&A] [answered]
I'm setting up warpx on HPC3, and I am stuck at the step on your readthedocs where I call: 

bash $HOME/src/warpx/Tools/machines/hpc3-uci/install_gpu_dependencies.sh

It has an issue with this line in adios2:

error: invalid use of undefined type 'struct fd_set'
    6 |   (void)sizeof(((struct fd_set *)0)->fds_bits);
      |                                    ^~

When I try to look for modules of adios2 to push to my setup, I find nothing.
Is there something wrong with the setup instructions, or maybe some prior step I need before I follow the instructions on the readthedocs?

This could also be a HPC3 issue, so I've messaged them as well.

**Comment by Aquios7** [comment, marked as answer]
I found that the git makefiles for the extra dependencies tried to install into root, which isn't allowed on my account. adjusting these makefiles to install in my own directory has fixed the problem of install so far.

**Reply by ax3l** [reply]
Glad you found a solution @Aquios7!

So does the `Tools/machines/hpc3-uci/install_gpu_dependencies.sh` script need an update? Can you pls share the exact error you encountered in an issue so we can update it in the documentation?

**Reply by Aquios7** [reply]
I don't believe there was a major issue with the script anymore. There was a single fi at line 29 that would make the script crash, but when it was removed everything worked as intended with the new instruction I used.

**Comment by ax3l** [comment]
cc @erny123 @floresv299 who also use this system.

@Aquios7 thank you for your report! Can you please share the changes you made to the docs of HPC3 https://warpx.readthedocs.io/en/latest/install/hpc/hpc3.html with us so we can update the documentation? Please feel free to post a pull request directly! :pray:
